crates/fuel-core/src/state/historical_rocksdb.rs

```diff
@@ -246,41 +275,133 @@ where
         Ok(())
     }
 
-    fn oldest_changes_height(&self) -> StorageResult<Option<u64>> {
-        let oldest_height = self
+    fn multiversion_changes_heights(
+        &self,
+        direction: IterDirection,
+        check_v1: bool,
+    ) -> (Option<StorageResult<u64>>, Option<StorageResult<u64>>) {
+        let v2_changes = self
             .db
-            .iter_all_keys::<ModificationsHistory<Description>>(Some(
-                IterDirection::Forward,
-            ))
-            .next()
-            .transpose()?;
+            .iter_all_keys::<ModificationsHistoryV2<Description>>(Some(direction))
+            .next();
+        let v1_changes = check_v1
+            .then(|| {
+                self.db
+                    .iter_all_keys::<ModificationsHistoryV1<Description>>(Some(direction))
+                    .next()
+            })
+            .flatten();
+
+        (v2_changes, v1_changes)
+    }
+
+    fn oldest_changes_height(&self) -> StorageResult<Option<u64>> {
+        let modifications_history_migration_in_progress = self.is_migration_in_progress();
+
+        let (v2_oldest_height, v1_oldest_height) = self.multiversion_changes_heights(
+            IterDirection::Forward,
+            modifications_history_migration_in_progress,
+        );
+
+        let v2_oldest_height = v2_oldest_height.transpose()?;
+        let v1_oldest_height = v1_oldest_height.transpose()?;
+
+        let oldest_height = match (v1_oldest_height, v2_oldest_height) {
+            (None, v2) => v2,
+            (v1, None) => v1,
+            (Some(v1), Some(v2)) => Some(v1.min(v2)),
+        };
         Ok(oldest_height)
     }
 
     #[cfg(test)]
     // TODO: This method doesn't work properly because of
     //  https://github.com/FuelLabs/fuel-core/issues/2095
     fn rollback_last_block(&self) -> StorageResult<u64> {
-        let latest_height = self
-            .db
-            .iter_all_keys::<ModificationsHistory<Description>>(Some(
-                IterDirection::Reverse,
-            ))
-            .next()
-            .ok_or(DatabaseError::ReachedEndOfHistory)??;
+        let modifications_history_migration_in_progress = self.is_migration_in_progress();
```

Comment:
The method would still not work properly after this PR, because the migration is not triggered at start. 
I would prefer to remove the todo in #2260